dhis-2/dhis-web-api-test/src/test/java/org/hisp/dhis/webapi/controller/trigramsummary/TrigramSummaryControllerTest.java

```diff
@@ -0,0 +1,254 @@
+        programInstance.setAutoFields();
+        programInstance.setEnrollmentDate( new Date() );
+        programInstance.setIncidentDate( new Date() );
+        programInstance.setStatus( ProgramStatus.COMPLETED );
+        programInstance.setFollowup( true );
+        manager.save( programInstance );
+    }
+
+    @Test
+    void getTrigramIndexSummaryWhenNoIndexesAreCreated()
+    {
+
+        when( trackedEntityAttributeTableManager.getAttributeIdsWithTrigramIndexCreated() )
+            .thenReturn( new ArrayList<>() );
+        TrigramSummary trigramSummary = controller.getTrigramSummary( new HashMap<>() );
+
+        assertNotNull( trigramSummary );
+
+        Set<String> expected = Set.of( "\"AttributeA\"", "\"AttributeB\"", "\"AttributeF\"" );
+        assertAttributeList( trigramSummary.getIndexableAttributes(), expected );
+
+        Set<String> expected2 = Set.of();
+        assertAttributeList( trigramSummary.getIndexedAttributes(), expected2 );
+
+        Set<String> expected3 = Set.of();
+        assertAttributeList( trigramSummary.getObsoleteIndexedAttributes(), expected3 );
+
+    }
+
+    @Test
+    void getTrigramIndexSummaryWithOneIndexAlreadyCreated()
+    {
+
+        when( trackedEntityAttributeTableManager.getAttributeIdsWithTrigramIndexCreated() )
+            .thenReturn( List.of( teaB.getId() ) );
+        TrigramSummary trigramSummary = controller.getTrigramSummary( new HashMap<>() );
+
+        assertNotNull( trigramSummary );
+
+        Set<String> expected = Set.of( "\"AttributeA\"", "\"AttributeF\"" );
+        assertAttributeList( trigramSummary.getIndexableAttributes(), expected );
+
+        Set<String> expected2 = Set.of( "\"AttributeB\"" );
+        assertAttributeList( trigramSummary.getIndexedAttributes(), expected2 );
+
+        Set<String> expected3 = Set.of();
+        assertAttributeList( trigramSummary.getObsoleteIndexedAttributes(), expected3 );
+
+    }
+
+    @Test
+    void getTrigramIndexSummaryWithAnObsoleteIndex()
+    {
+
+        when( trackedEntityAttributeTableManager.getAttributeIdsWithTrigramIndexCreated() )
+            .thenReturn( List.of( teaB.getId(), teaC.getId() ) );
+        TrigramSummary trigramSummary = controller.getTrigramSummary( new HashMap<>() );
+
+        assertNotNull( trigramSummary );
+
+        Set<String> expected = Set.of( "\"AttributeA\"", "\"AttributeF\"" );
+        assertAttributeList( trigramSummary.getIndexableAttributes(), expected );
+
+        Set<String> expected2 = Set.of( "\"AttributeB\"" );
+        assertAttributeList( trigramSummary.getIndexedAttributes(), expected2 );
+
+        Set<String> expected3 = Set.of( "\"AttributeC\"" );
+        assertAttributeList( trigramSummary.getObsoleteIndexedAttributes(), expected3 );
+
+    }
+
+    private static void assertAttributeList( List<ObjectNode> attributes, Set<String> expected )
+    {
+        assertEquals( expected.size(), attributes.size() );
```

Comment:
you can wrap these 2 assertions in https://junit.org/junit5/docs/5.8.2/api/org.junit.jupiter.api/org/junit/jupiter/api/Assertions.html#assertAll(org.junit.jupiter.api.function.Executable...)

this way if the expected sizes dont match we get a more useful message also checking the other assertion. Otherwise, I usually have to rerun the test, commenting out the size to see the contents and figure out why the sizes dont add up.